use ember-concurrency to handle multiple signup requests

sbatson5 opened the pull request:
# What's in this PR?

We use `ember-concurrency` in a couple of places in our app already. We should leverage tasks from `ember-concurrency` to solve the problem of trying to submit data multiple times.  This PR addresses the signup form and ensures the user can't spam the submit button.
## References

Progress on: #507 
Progress on: #336

sbatson5 commented on app/components/signup-form.js:
```diff
@@ -67,20 +53,18 @@ export default Component.extend({
-  }
+
+    yield promise;
+  }).drop()
```
From the `ember-concurrency` documentation:

> The drop modifier `drops` tasks that are `.performed` while another is already running. Dropped tasks' functions are never even called.

JoshSmith replied:
@sbatson5 I'm a little confused by this. Does this mean that a request will still continue on to the server?

sbatson5 replied:
@JoshSmith Yes. It means that any additional clicks (after the first one) will be "dropped." So we should never see more than 1 request going out.

sbatson5 commented on app/components/signup-form.js:
```diff
@@ -25,19 +22,14 @@ export default Component.extend({
 
-  init() {
-    this._super(...arguments);
-    this.set('user', this.get('store').createRecord('user'));
```
Moved this to the route so we don't have to inject the store in the component anymore.

sbatson5 commented:
@JoshSmith let me know if this looks ready and I will squash

JoshSmith commented on tests/unit/routes/signup-test.js:
```diff
@@ -1,11 +1,39 @@
   assert.ok(route);
 });
+
+test('it renders erros through controller', function(assert) {
```
`erros` needs fixed.

JoshSmith commented on app/templates/signup.hbs:
```diff
@@ -1,3 +1,5 @@
 {{title "Sign up"}}
 
-{{signup-form}}
+{{signup-form user=model
```
Do we want to place `user=model` on a separate line? Alpha order these?

sbatson5 replied:
Can do!

sbatson5 commented:
That's a good question @begedin. Not sure how we could test concurrency in something like this. And even if we did, aren't we just testing `ember-concurrency` and not our code?

Also, rebased and squashed.
